src/main/java/org/jvnet/hudson/update_center/Plugin.java

```diff
@@ -69,43 +67,23 @@
      * Latest version of this plugin.
      */
     public final HPI previous;
-    /**
-     * Confluence page of this plugin in Wiki.
-     * Null if we couldn't find it.
-     */
-    public final RemotePage page;
-
-    /**
-     * Confluence labels for the plugin wiki page.
-     * Null if wiki page wasn't found.
-     */
-    private String[] labels;
-    private boolean labelsRead = false;
-
-    /**
-     * Deprecated plugins should not be included in update center.
-     */
-    private boolean deprecated = false;
 
     private final SAXReader xmlReader;
-    private final ConfluencePluginList cpl;
 
     /**
      * POM parsed as a DOM.
      */
-    private final Document pom;
+    private Document pom;
 
-    public Plugin(String artifactId, HPI latest, HPI previous, ConfluencePluginList cpl) throws IOException {
+    public Plugin(String artifactId, HPI latest, HPI previous) throws IOException {
         this.artifactId = artifactId;
         this.latest = latest;
         this.previous = previous;
         this.xmlReader = createXmlReader();
         this.pom = readPOM();
```

Comment:
`pom` should be initialized lazily as you are porting 97e070138180cba9a728617ca468be5fe0f98813@jenkins-infra/backend-update-center2